### PR TITLE
Add astropy.units support to config layer.

### PR DESCRIPTION
**Pull request by jmeyers314**
Starting to add astropy Quantity/Unit support to GalSim config layer...  WIP, but could be worth a quick look at what's here so far.

### CODE REVIEW

**Comment by jmeyers314** [comment]
I think this is ready now.  I added `Quantity` and `Unit` as config-layer value types.  These can be instantiated as described in #1194 .  I then added support for a number of concrete config classes, including:

* `Airy`, `Kolmogorov`, `VonKarman`, `OpticalPSF` for wavelengths, Fried parameters, telescope diameters.
* Bandpass units
* SED units and flux densities
* DCR conditions (pressure/temperature/H2O_pressure)

I think there are a lot of places where this will particularly benefit ImSim.

**Comment by cwwalter** [comment]
This is great Josh.  

FYI, here is an old imSim issue about this:

https://github.com/LSSTDESC/imSim/issues/48

File: docs/config_values.rst [hunk 1]
```diff
@@ -571,6 +573,49 @@ Options are:
...
+
+These represent `astropy.units.Quantity` values, which are a combination of a float and a unit (specifically, an `astropy.units.Unit`).
+
+Options are:
+
+* An `astropy.units.Quantity` value (e.g. '8.7*u.m', where 'u' is the `astropy.units` module)
+* A string interpretable by `astropy.units.Quantity` (e.g. '8.7 m')
+* A dict with:
+
+    * ``type`` = *str* (required)  There is only one valid option:
```
Comment (rmjarvis) [inline docs/config_values.rst:587]: There seem to be two valid options.

File: docs/config_values.rst [hunk 2]
```diff
@@ -571,6 +573,49 @@ Options are:
...
+
+These represent `astropy.units.Unit` values.
+
+Options are:
+
+* An `astropy.units.Unit` value (e.g. 'u.m', where 'u' is the `astropy.units` module)
+* A string interpretable by `astropy.units.Unit` (e.g. 'm')
+* A dict with:
+
+    * ``type`` = *str* (required)  There is only one valid option:
```
Comment (rmjarvis) [inline docs/config_values.rst:609]: Here too.

**Review by rmjarvis — APPROVED** [review]
Looks great Josh!  Just a couple minor doc errors.

File: docs/config_values.rst (hunk 1 above)
Comment (jmeyers314) [reply docs/config_values.rst:587]: Looks like I mimicked the `sky_value` section, which also technically has two valid options.  I'll fix that one too.

**Pull request merged**